### PR TITLE
Feat/add experimental testing matrix failure options

### PR DESCRIPTION
**Pull request by chelseapinka**
# Making Dev-Next and Node18 experimental features for testing matrix gating.

# Checklist

- [x] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [x] New functions/types have been exported in `index.ts`, if applicable.
- [x] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [x] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: .github/workflows/e2e-node.yml [hunk 1]
```diff
@@ -15,11 +15,17 @@ jobs:
...
       matrix:
         os: [ubuntu-latest]
-        node-version: [16.x]
-        # NSS does not support static client registration, which we rely on for testing.
-        environment-name: ["ESS Dev-Next"]
+        node-version: ["14.x", "16.x"]
+        environment-name: ["ESS PodSpaces", "ESS PodSpaces Next"]
+        experimental: [false]
+        include:
+          - version: "18.x"
```
Comment (NSeydoux) [inline .github/workflows/e2e-node.yml:25]:
```suggestion
          - version: "16.x"
```
We currently don't support Node 18 at all

File: .github/workflows/audit.yml [hunk 2]
```diff
@@ -7,6 +7,6 @@ on:
 
 jobs:
   audit:
-    uses: inrupt/javascript-style-configs/.github/workflows/reusable-audit.yml@v1.1.0
+    uses: inrupt/typescript-sdk-tools/.github/workflows/reusable-audit.yml@v1.1.0
```
Comment (ThisIsMissEm) [inline .github/workflows/audit.yml:10]:
```suggestion
    uses: inrupt/typescript-sdk-tools/.github/workflows/reusable-audit.yml@v1.4.2
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ env:
   CI: true
 jobs:
   lint:
-    uses: inrupt/javascript-style-configs/.github/workflows/reusable-lint.yml@v1.1.0
+    uses: inrupt/typescript-sdk-tools/.github/workflows/reusable-lint.yml@v1.1.0
```
Comment (ThisIsMissEm) [inline .github/workflows/ci.yml:9]:
```suggestion
    uses: inrupt/typescript-sdk-tools/.github/workflows/reusable-lint.yml@v1.4.2
```

File: .github/workflows/e2e-node.yml [hunk 4]
```diff
@@ -15,11 +15,17 @@ jobs:
...
+            experimental: true
```
Comment (ThisIsMissEm) [inline .github/workflows/e2e-node.yml:26]:
```suggestion
```

This is the better suggestion for what @NSeydoux says above.

File: .github/workflows/e2e-node.yml [hunk 5]
```diff
@@ -15,11 +15,17 @@ jobs:
...
       name: ${{ matrix.environment-name }}
     strategy:
+      fail-fast: true
       matrix:
         os: [ubuntu-latest]
-        node-version: [16.x]
-        # NSS does not support static client registration, which we rely on for testing.
-        environment-name: ["ESS Dev-Next"]
+        node-version: ["14.x", "16.x"]
+        environment-name: ["ESS PodSpaces", "ESS PodSpaces Next"]
```
Comment (ThisIsMissEm) [inline .github/workflows/e2e-node.yml:22]:
```suggestion
        environment-name: ["ESS PodSpaces"]
```

ESS PodSpaces Next no longer exists. There is only ESS PodSpaces now

File: .github/workflows/e2e-node.yml [hunk 6]
```diff
@@ -15,11 +15,17 @@ jobs:
...
+        experimental: [false]
+        include:
+          - version: "18.x"
+            experimental: true
+          - environment-name: "ESS Dev-Next"
+            experimental: true
```
Comment (ThisIsMissEm) [inline .github/workflows/e2e-node.yml:28]:
```suggestion
            node-version: ["14.x", "16.x"]
            experimental: true
```

File: e2e/node/e2e.test.ts [hunk 7]
```diff
@@ -24,13 +24,23 @@
 
 import { describe, it, expect, beforeEach, afterEach } from "@jest/globals";
 import { Session } from "@inrupt/solid-client-authn-node";
-import { getNodeTestingEnvironment } from "@inrupt/internal-test-env";
+import { setupEnv, getNodeTestingEnvironment } from "@inrupt/internal-test-env";
```
Comment (ThisIsMissEm) [inline e2e/node/e2e.test.ts:27]:
```suggestion
import { getNodeTestingEnvironment } from "@inrupt/internal-test-env";
```

See next comment

File: e2e/node/e2e.test.ts [hunk 8]
```diff
@@ -24,13 +24,23 @@
...
+// Load environment variables from .env.test.local if available:
+// config({
+//   path: __dirname,
+//   // In CI, actual environment variables will overwrite values from .env files.
+//   // We don't need warning messages in the logs for that:
+//   silent: process.env.CI === "true",
+// });
+
+setupEnv();
+
```
Comment (ThisIsMissEm) [inline e2e/node/e2e.test.ts:43]:
```suggestion
```

This should be done by `e2e/node/jest.setup.ts`

File: e2e/node/e2e.test.ts [hunk 9]
```diff
@@ -95,21 +105,24 @@ const invalidCredentialClaims = {
   },
 };
 
-const env = getNodeTestingEnvironment({
+const {
+  idp: oidcIssuer,
+  vcProvider,
+  clientCredentials,
+} = getNodeTestingEnvironment({
```
Comment (ThisIsMissEm) [inline e2e/node/e2e.test.ts:112]: I thought we agreed on not destructuring this?

**Review by ThisIsMissEm — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-node.yml (hunk 6 above)
Comment (NSeydoux) [reply .github/workflows/e2e-node.yml:26]: Note that this isn't possible in the `include` section of the matrix: https://github.com/inrupt/solid-client-vc-js/actions/runs/3542226441

File: e2e/node/e2e.test.ts (hunk 9 above)
Comment (NSeydoux) [reply e2e/node/e2e.test.ts]: https://github.com/inrupt/solid-client-vc-js/pull/531/commits/f9acfd45b54ae92f31ede14e7e270d90d73e4bbc

**Review by ThisIsMissEm — APPROVED** [review]

**Review by ThisIsMissEm — CHANGES_REQUESTED** [review]
This missed the concurrency stuff, I'll add that in now.

**Review by ThisIsMissEm — APPROVED** [review]
Ignore me.. seems github wasn't showing me the true, good to go!

**Pull request merged**